src/libstore/binary-cache-store.cc

```diff
@@ -290,8 +310,23 @@ bool BinaryCacheStore::isValidPathUncached(const Path & storePath)
 void BinaryCacheStore::narFromPath(const Path & storePath, Sink & sink)
 {
     auto info = queryPathInfo(storePath).cast<const NarInfo>();
+    std::shared_ptr<std::string> nar;
+
+#if ENABLE_IPFS
```

Comment:
[As commented](https://github.com/NixOS/nix/issues/859#issuecomment-273931467), downloading from IPFS can open the door for new attacks. I will recommend adding a `download-from-ipfs` flag as well which is disabled by default.